[Chu Yi-Ting] iP

chuyiting opened the pull request.

chuyiting committed:
Level1 - Echo built

chuyiting committed:
Level2 - Add & List functionality built

chuyiting committed:
Level3 Mark as Done built

chuyiting committed:
Level 4 - Todos, Deadlines, events built

chuyiting committed:
Test added

chuyiting committed:
Exceptions of possible incomplete inputs

chuyiting committed:
Index out of bound customized

chuyiting committed:
Task Deletion built

chuyiting committed:
Refactor files related to Duke

Classes related to Duke do not meet the coding standard.

A package named Duke is created to refactor the classes.

Classes related to Duke should all be packaged into a single package, by
doing so, classes related to Duke is encapsulated.

Refer to Package tutorial on GeeksForGeeks
https://www.geeksforgeeks.org/packages-in-java/

chuyiting committed:
Level-7 Save

Use a csv file as database.

chuyiting committed:
Level8. Dates and Times

Define standard format for Date.

chuyiting committed:
Revert "Level8. Dates and Times"

This reverts commit 1293c7a344b005def5cea5b02f813d70de4cd603.

chuyiting committed:
A-MoreOOP

Scalability of the program was not good because of the weak abstraction
design.

Refactor i/o related classes into ui package, encapsulate data storage
and decrease the dependency between each classes.

chuyiting committed:
Merge branch 'master' of https://github.com/chuyiting/ip

# Conflicts:
#	src/main/java/duke/DukeCommandMatcher.java
#	src/main/java/duke/tasks/SingletonTaskList.java
#	src/main/java/duke/utils/Constants.java
#	src/main/java/duke/utils/UtilFunction.java

chuyiting committed:
Add JUnit test

chuyiting committed:
A-Jar

Build Jar file

chuyiting committed:
Add Javadoc

chuyiting committed:
Add find functionality.

chuyiting committed:
Adjust to coding standard.

bchenghi commented on src/main/java/duke/DukeCommandMatcher.java:
```diff
@@ -0,0 +1,167 @@
+    }
+
+
+    public String matchCommand(String command) throws CommandNotFoundException, NullCommandException,
```
This method might be a little long?

chuyiting replied:
Yeah, it does not look nice, but because of the same behaviour that Duke would respond to all these exceptions, I decide to throw all of them and handle it together in the highest level. Thanks for the suggestion.

bchenghi commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,41 @@
+     * @throws IllegalStateException
+     * @see IllegalStateException
+     */
+    public static void loop(Storage database) throws IllegalStateException {
```
loop as a method name, could it be a little vague?

chuyiting replied:
Hmm, as I imagine a working application is something that is looping all the time until you shut it down. Any better suggestions for the name>

bchenghi commented on src/main/java/duke/utils/Constants.java:
```diff
@@ -0,0 +1,64 @@
+package duke.utils;
+
+
+public class Constants {
```
I think class names should be singular?

bchenghi commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,120 @@
+     * Real all the tasks from the database.
+     * @return the list of all the tasks from the database
+     */
+    public List<Task> readAll() {
```
Method name might be a little vague? Since the method returns a list.

chuyiting replied:
Hmm, yeah, I kind of follow the idea of CRUD (the read), as this method scans through the database, therefore I name it readAll. Any suggestions?

bchenghi commented on src/main/java/duke/utils/UtilFunction.java:
```diff
@@ -0,0 +1,80 @@
+        }
+    }
+
+    private static void lineSentence(String sentence) {
```
Perhaps method names can be verbs or actions?

chuyiting replied:
Hahaha, I spent few minutes pondering on the name of this method, as what it does is basically "auto-change-line" if the sentence is longer than the console (the same as wrapText). I did not know how to name it at that time, I guess ill follow the JavaFX convention. thanks for the suggestion

bchenghi reviewed (commented):
LGTM! Overall, I thought the coding standard was really good! There were no arrow heads and the code was easy to read. I have some minor suggestions, but other than that, I feel that it's good.

chuyiting committed:
Add checkstyle and reduce errors to 0

WhiteLio commented on src/main/java/duke/DukeCommandMatcher.java:
```diff
@@ -0,0 +1,190 @@
+ */
+public class DukeCommandMatcher {
+
+    private static final List<String> commandList = new ArrayList<>(Arrays.asList(Constants.LISTPATTERN,
```
this constant should be in all caps! `COMMAND_LIST`

chuyiting replied:
Ahh, forgot about it, thanks.

WhiteLio commented on src/main/java/duke/DukeCommandMatcher.java:
```diff
@@ -0,0 +1,190 @@
+     * Match the command to corresponding behavior.
+     * @param command
+     * @return message of implementation
+     * @throws CommandNotFoundException
+     * @throws NullCommandException
+     * @throws LackOfTimeException
+     * @throws NullCommandContentException
+     * @throws TaskOutOfBoundException
+     * @throws TaskNotSpecifyException
+     * @throws DateFormatException
```
you should specify when these exceptions are thrown!
eg `@throws CommandNotFoundException If the user command is not in the command list`

chuyiting replied:
Thanks for the remind, I forgot to leave a readable note for them. Will add soon.

WhiteLio commented on src/main/java/duke/DukeCommandMatcher.java:
```diff
@@ -0,0 +1,190 @@
+    }
+
+    private String handleExit() {
+        System.out.println("Farewell/再見/さようなら～～");
```
귀여워요 haha

chuyiting replied:
Good suggestion haha

WhiteLio commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,130 @@
+
+    private Task storageLineParser(String[] taskStr) {
+        Task task;
+        switch (taskStr[0]) {
+        case "T": {
+            task = new ToDo(taskStr[2], taskStr[1]);
+            break;
+        }
+        case "E": {
+            task = new Event(taskStr[1], taskStr[2], taskStr[3]);
+            break;
+        }
+        case "D": {
+            task = new Deadline(taskStr[1], taskStr[2], taskStr[3]);
+            break;
+        }
+        default: {
+            task = null;
+            break;
+        }
+        }
```
those curly braces are unnecessary! i think the code would look cleaner without them~

WhiteLio commented on src/main/java/duke/utils/Constants.java:
```diff
@@ -0,0 +1,64 @@
+    /**
+     * Exit pattern of the exit command for Duke.
+     */
+    public static final String EXITPATTERN = ("^(b|B)(y|Y)(e|E)$");
```
you should use underscores to separate words! also, associated constants (final variables) should be prefixed by a common type name
eg `PATTERN_EXIT`, `PATTERN_DONE`

chuyiting replied:
Ahh, right, that is a part of the coding standard, I forgot about it, yeah, maybe I'll also consider changing it into an Enum class, thanks.

WhiteLio reviewed (commented):
looks great! keep it up :)

hyngkng commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,48 @@
+    private String duration;
+
+    /**
+     * Default constructor for a new event object.
```
Shouldn't there be a space between the description and param section?

hyngkng reviewed (commented):
Overall I like your coding style and your code generally follows the coding standard. Just some nits to fix!

chuyiting committed:
Build gui for Duke

chuyiting committed:
Update GUI

chuyiting committed:
Update code with assertions.

chuyiting committed:
Merge pull request #1 from chuyiting/branch-assertion

Add assertions

chuyiting committed:
Refactor Cli class to conform to SLAP.

chuyiting committed:
Create gradle.yml

Start CI

chuyiting committed:
Merge pull request #2 from chuyiting/branch-CI

Create gradle.yml

chuyiting committed:
Update Windows and Linux dependency for JavaFX to pass CI.

chuyiting committed:
Merge branch 'master' of https://github.com/chuyiting/ip

chuyiting committed:
Delete IO redirection test

chuyiting committed:
Merge branch 'branch-codeQuality'

chuyiting committed:
Delete repeated yml file.

chuyiting committed:
Build natural date interpreter.

chuyiting committed:
Fix Junit testing.

chuyiting committed:
Fix test checkStyle.

chuyiting committed:
Fix GUI bugs

chuyiting committed:
Fix style problem

chuyiting committed:
Fix input diable bug.

The listener was still enabled after "bye" command.

The listener is disabled after "bye".

chuyiting committed:
Add event duration function.

chuyiting committed:
Refactor parser

chuyiting committed:
Add tag GUI

chuyiting committed:
Fix few bugs.

chuyiting committed:
Fix checkstyle.

chuyiting committed:
Fix checkstyle

chuyiting committed:
Add error hightlighting.

chuyiting committed:
Fix checkstyle.

chuyiting committed:
Add Ui picture

chuyiting committed:
Set theme jekyll-theme-architect

chuyiting committed:
Update User Guide

chuyiting committed:
Merge branch 'master' of https://github.com/chuyiting/ip

chuyiting committed:
Fix small find bugs

chuyiting committed:
Fix minor textfield bug.

Add command instruction to README.

chuyiting committed:
Fix css and javaFX error messages.
